node/flatpak_node_generator/providers/npm.py

```diff
@@ -79,11 +80,17 @@ def _process_packages_v1(
             elif version_url.scheme == 'file':
                 source = LocalSource(path=version_url.path)
             else:
-                integrity = Integrity.parse(info['integrity'])
+                integrity = None
+                if 'integrity' in info:
+                    integrity = Integrity.parse(info['integrity'])
+
                 if 'resolved' in info:
-                    source = ResolvedSource(
-                        resolved=info['resolved'], integrity=integrity
-                    )
+                    if info['resolved'].startswith('git+'):
```

Comment:
`info['resolved']` should probably be in an intermediate variable so the same string constant isn't copy-pasted in 3 places. (this is also something the rest of the code admittedly does poorly.)

Reply:
Fixed, thanks.